### PR TITLE
Add Sprite Clipboard Functions

### PR DESCRIPTION
**Pull request by time-killer-games**
Adds these functions:
- void clipboard_set_sprite(int ind, unsigned subimg)
- int clipboard_get_sprite(bool precise)
- bool clipboard_has_sprite()

### CODE REVIEW

**Comment by time-killer-games** [comment]
@JoshDreamland ready for review

**Comment by time-killer-games** [comment]
@JoshDreamland ready for review

**Comment by time-killer-games** [comment]
@JoshDreamland ready for review

File: ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp [hunk 1]
```diff
@@ -694,4 +703,79 @@ bool clipboard_has_text()
...
+      int base = (y * bih.biWidth + x) * 4;
+      dst[j++] = src[base];
+      dst[j++] = src[base + 1];
+      dst[j++] = src[base + 2];
+      dst[j++] = (i != n / 4) ? src[base + 3] : 255;
+    }
+  }
+
+  enigma::spritestructarray_reallocate();
+  enigma::sprite *spr = enigma::spritestructarray[enigma::sprite_idmax] = new enigma::sprite();
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp]: Don't rely on sprite internals; they are likely to change. Use `sprite_create_empty` (or whatever it was called) to get a new sprite, then just feed it the translated bitmap data.

File: ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp [hunk 2]
```diff
@@ -694,4 +703,79 @@ bool clipboard_has_text()
...
+  free(src);
+}
+
+int clipboard_get_sprite(bool precise, bool transparent, bool smooth, int x_offset, int y_offset) {
+  if (!clipboard_has_sprite()) return -1;
+  OpenClipboard(NULL);
+  HBITMAP hBitmap = (HBITMAP)GetClipboardData(CF_BITMAP);
+  CloseClipboard();
+
+  BITMAPINFOHEADER bih = {0};
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSwindow.cpp]: Lock the bitmap data, and pass it to `image_decode_bmp`. You will need to construct a string or string view from the data and its size (eg, `string(clip_data, GlobalSize(clip_data))`).

**Pull request closed without merging**